### PR TITLE
312/feat/trainer can delete own templates

### PR DESCRIPTION
**Pull request by kodtld**

### CODE REVIEW

File: src/app/api/template/route.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+} from '@/lib/response/responseUtil';
+import { prisma } from '@/lib/prisma';
+import { handleCommonErrors } from '@/lib/response/errorUtil';
+import { getServerAuthSession } from '@/lib/auth';
+import { isTrainerOrAdmin, hasTemplateDeleteRights } from '@/lib/auth-utils';
+import { translator } from '@/lib/i18n';
+import { templateDeleteSchema } from '@/lib/zod/templates';
+
+export async function DELETE(request: NextRequest) {
+  console.log('template delete');
```
Comment (akskokki) [inline src/app/api/template/route.ts:15]: Debugging console.log call needs to be removed

File: src/app/api/template/route.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  try {
+    const { t } = await translator('api');
+    const { user } = await getServerAuthSession();
+    const reqData = await request.json();
+    const template = templateDeleteSchema.parse(reqData);
+
+    const template_exists = await prisma.template.findFirst({
```
Comment (akskokki) [inline src/app/api/template/route.ts:22]: variable doesn't follow camelCase convention

File: src/app/api/template/route.test.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import { MessageType, StatusCodeType } from '@/lib/response/responseUtil';
+import { DELETE } from './route';
+import { prisma, clearDatabase } from '@/lib/prisma';
+import { NextRequest } from 'next/server';
+import { createMocks } from 'node-mocks-http';
+import { Role } from '@prisma/client';
+import { getServerAuthSession } from '@/lib/auth';
+
+const TrainerUser1 = {
```
Comment (akskokki) [inline src/app/api/template/route.test.ts:9]: variables TrainerUser1 and TrainerUser2 don't follow camelCase convention

File: src/components/DeleteTemplate/DeleteTemplateButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    renderWithTheme(<DeleteTemplateButton templateId="1" lang="en" />);
+    const buttonElement = screen.getByTestId('DeleteTemplateButton');
+    expect(buttonElement).toBeInTheDocument();
+    expect(buttonElement).toHaveTextContent(
+      'DeleteTemplateButton.button.delete'
+    );
+  });
+});
+
+test('opens the popup when the button is clicked', () => {
```
Comment (akskokki) [inline src/components/DeleteTemplate/DeleteTemplateButton.test.tsx:45]: I think these should be under the describe('DeleteTemplateButton') function above?

**Review by akskokki — APPROVED** [review]
A few small formatting things that would be good to fix before merging, otherwise looks good!

File: src/app/api/template/route.ts (hunk 1 above)
Comment (kodtld) [reply src/app/api/template/route.ts]: Fixed

File: src/app/api/template/route.ts (hunk 2 above)
Comment (kodtld) [reply src/app/api/template/route.ts]: Fixed

File: src/app/api/template/route.test.ts (hunk 3 above)
Comment (kodtld) [reply src/app/api/template/route.test.ts]: Fixed

File: src/components/DeleteTemplate/DeleteTemplateButton.test.tsx (hunk 4 above)
Comment (kodtld) [reply src/components/DeleteTemplate/DeleteTemplateButton.test.tsx]: Fixed

**Pull request merged**